deku-derive/src/macros/deku_write.rs

```diff
@@ -1,54 +1,93 @@
-use crate::{DekuFieldReceiver, DekuReceiver, EndianNess};
+use crate::macros::{gen_ctx_types_and_arg, gen_field_args, gen_id_args, gen_struct_destruction};
+use crate::{DekuData, FieldData};
 use darling::ast::{Data, Fields};
 use proc_macro2::TokenStream;
 use quote::quote;
 
-pub(crate) fn emit_deku_write(input: &DekuReceiver) -> Result<TokenStream, darling::Error> {
-    assert!(
-        input.id_bytes.is_none(),
-        "dev error: `id_bytes` should be None, use `id_bits` to get size"
-    );
-
+pub(crate) fn emit_deku_write(input: &DekuData) -> Result<TokenStream, syn::Error> {
     match &input.data {
         Data::Enum(_) => emit_enum(input),
         Data::Struct(_) => emit_struct(input),
     }
 }
 
-fn emit_struct(input: &DekuReceiver) -> Result<TokenStream, darling::Error> {
+fn emit_struct(input: &DekuData) -> Result<TokenStream, syn::Error> {
     let mut tokens = TokenStream::new();
 
     let (imp, ty, wher) = input.generics.split_for_impl();
 
     let ident = &input.ident;
     let ident = quote! { #ident #ty };
 
+    // TODO: Replace `expect` with `Err`
     let fields = input
         .data
         .as_ref()
         .take_struct()
         .expect("expected `struct` type");
 
-    let field_writes = emit_field_writes(input, &fields, Some(quote! { self. }))?;
+    let field_writes = emit_field_writes(input, &fields, None)?;
     let field_updates = emit_field_updates(&fields, Some(quote! { self. }))?;
 
-    tokens.extend(quote! {
-        impl #imp core::convert::TryFrom<#ident> for BitVec<Msb0, u8> #wher {
-            type Error = DekuError;
+    /*
+    NOTE:
+    Because the requirement by `ctx`, we need to deconstruct first.
```

Comment:
can you clarify the comment? why do we need to deconstruct? scoping/naming issue?

Reply:
it's a naming issue. for example:
```rust
#[derive(DekuWrite)]
struct A {
    a: u8,
    #[deku(ctx = "a")]
    b: Subtype,
    #[deku(writer = "write((a, b))")]
    c: u8
}
// expand to
fn write(...) -> ... {
    self.b.write(a); // <-- `a` not found, should be &self.a
    write((a, b)); // <-- `a, b` not found, should be &self.a, &self.b
}
```

Reply:
Ah! Gotcha.